### ISSUE TITLE
Wrong hint about needless_range_loop

### ISSUE THREAD

**Issue by BusyJay** [Manishearth/rust-clippy#878]
Clippy warns about following snippet:

```
let v = vec![5; 10];
for i in 0..v.len() - 2 {
    println!("{}", v[i + 2] - v[i]);
}
```

It suggest me to use `enumerate`. But I don't think I can use `enumerate` in this situation.

**Comment by Manishearth** [comment]
`.enumerate().first(v.len()-2)`

It looks more complicated but you don't have to worry about bounds checks.

**Comment by BusyJay** [comment]
No, I need to access two items at different position. In this case, I need to access `i + 2` and `i`.

**Comment by oli-obk** [comment]
I think the proper way would be

```rust
for (&a, &b) in v[2..].iter().zip(&v) {
    println!("{}", a - b);
}
```

**Comment by Manishearth** [comment]
(While there are ways of doing this with iterators, the hint should still be improved)

**Comment by BusyJay** [comment]
@oli-obk I think my version is more clean and simple. :)

**Comment by oli-obk** [comment]
@BusyJay you need to specify the `2` two times ;) mine is less readable for sure, but that's mainly b/c we don't have good ways to zip like `for (&a, &b) in (&v[2..], &v) {}`

**Comment by BusyJay** [comment]
Maybe `windows` is a good choice in this situation.

**Comment by y21** [comment]
Is this issue still relevant today? It doesn't seem to lint on the code provided anymore: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ac000dbc5a08cd445b4babe76f6176bf

**Comment by Manishearth** [comment]
Yeah that lint has gotten tons of improvement but we probably had duplicate issues filed. Nice catch!
